src/tests/ftest/aggregation/aggregation_throttling.py

```diff
@@ -0,0 +1,151 @@
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+
+import time
+from ior_test_base import IorTestBase
+from ior_utils import IorCommand, IorMetrics
+from apricot import skipForTicket
+
+class DaosAggregationThrottling(IorTestBase):
+    # pylint: disable=too-many-ancestors
+    """Test class Description:
+
+       Verify ior performance during aggregation throttling
+
+    :avocado: recursive
+    """
+
+    @skipForTicket("DAOS-")
+    def test_aggregation_throttling(self):
+        """Jira ID: DAOS-3749
+
+        Test Description:
+            Verify the ior throttling during aggregation
+            in the background is affecting the ior
+            performance only by +/- 10%
+        Use case:
+            Create a pool and container
+            Disable the aggregation
+            Run ior with same file option
+            Capture the initial ior performance.
+            Run ior the second time with same file option, so the
+            logical partitioning can be overwritten.
+            Enable the aggregation and wait for 90 seconds
+            Now as the aggregation is running in the background, run
+            ior again so both aggregation and ior runs in parallel
+            Capture the ior performance now and verify that it is
+            +/- 10% of the initial performance.
+            Also, verify the aggregation reclaimed the space used by
+            second ior.
+
+        :avocado: tags=all,pr,hw,large,aggregate,daosio,aggregatethrottling
+        """
+
+        # Create pool and container
+        self.update_ior_cmd_with_pool()
+
+        # Disable the aggregation
+        self.pool.set_property("reclaim", "disabled")
+
+        # Run ior with -k option to retain the file and not delete it
+        out = self.run_ior_with_pool()
+        metric_before_aggregate = IorCommand.get_ior_metrics(out)
+
+        # Run ior the second time on the same pool and container, so another
+        # copy of the file is inserted in DAOS.
+        out = self.run_ior_with_pool(update=False)
+
+        # wait 90 seconds for files to get old enough for aggregation
+        self.log.info("Waiting for 90 seconds for aggregation to start")
+        time.sleep(90)
+        # Enable the aggregation
+        self.pool.set_property("reclaim", "time")
+
+        # Run ior the third time while the aggregation of first two runs
+        # are running in the background.
+        out = self.run_ior_with_pool(update=False)
+        metric_after_aggregate = IorCommand.get_ior_metrics(out)
+
+        # When DAOS-5057 is fixed, adjust the percentage. For now, 
+        # keep it at 30 %
+        expected_perf_diff = 30.0
+
+        self.verify_performance(metric_before_aggregate,
+                                metric_after_aggregate,
+                                0, # write_perf
+                                expected_perf_diff) # 10% perf difference
```

Comment:
For Line 91 above, the perf diff mentioned is 30%, so may be for the comment in this line you might want to change it to 30% from 10%.

I did saw the ticket mentions 10% perf diff and later there is a comment by you that write perf is 30% diff. Is this acceptable number as per developer or should there be a parallel ticket to handle this issue?

Reply:
JIRA # DAOS-5057 is there to look into the performance issue. But to get this ticket to CI, 30% diff is acceptable for now per Johann's comment. Will need to revisit based on DAOS-5057 resolution.